### ISSUE TITLE
Passing null for array of blittable types results in NullReferenceException

### ISSUE THREAD

**Issue by elinor-fung** [dotnet/runtimelab#1051]
Generated code doesn't handle `null` when marshalling an array of blittable types.
```C#
[GeneratedDllImport(NativeExportsNE_Binary, EntryPoint = "sum_int_array")]
public static partial int Sum(int[] values, int numValues);
...
NativeExportsNE.Sum(null, 0);
```

In the generated code, we pass null to `MemoryMarshal.GetArrayDataReference` throws `NullReferenceException`:
```C#
fixed (int* __values_gen_native = &System.Runtime.InteropServices.MemoryMarshal.GetArrayDataReference(values))
    __retVal = Sum__PInvoke__(__values_gen_native, numValues);
```

cc @AaronRobinsonMSFT @jkoritzinsky

**Comment by jkotas** [comment]
~~Also, this should generate just `fixed (int* __values_gen_native = values)` for concrete arrays.~~

**Comment by jkoritzinsky** [comment]
We can't just do `fixed (int* __values_gen_native = values)` because that results in a `null` pointer for a zero-length array, which breaks P/Invokes to libraries like GDI+.

**Comment by AaronRobinsonMSFT** [comment]
@jkoritzinsky I recall this being an issue we wanted to fix but given the compat burden we decided not to do that. We should have a conversation around the GDI+ API surface and see if it is something we want to support. I'm not convinced a single API surface should force this decision.

**Comment by jkotas** [comment]
These bogus null pointer checks are not unusual in the Windows APIs. For example here: https://github.com/dotnet/runtime/blob/01b7e73cd378145264a7cb7a09365b41ed42b240/src/libraries/System.Private.CoreLib/src/System/Globalization/CompareInfo.Icu.cs#L733 or here https://github.com/dotnet/runtime/pull/52030/files#diff-4b9bcd238c75b08d184c630e8e90372464f848bbb2ed0466c0c3bafe74ee27b7R23

The eventual marshaling of Spans will hit similar problem.

**Comment by jkoritzinsky** [comment]
Fixed by #1063
